src/FakeItEasy/Creation/CreationResult.cs

```diff
@@ -6,11 +6,14 @@ namespace FakeItEasy.Creation
     using System.Text;
     using FakeItEasy.Core;
 
-    internal abstract class CreationResult
+    internal abstract class CreationResult(int status)
     {
+        private const int SuccessfulCreation = 0;
+        private const int FailedCreation = 1;
+
         public static CreationResult Untried { get; } = new UntriedCreationResult();
 
-        public abstract bool WasSuccessful { get; }
+        public bool WasSuccessful => status == SuccessfulCreation;

```

Comment:
Mostly I wanted to simplify the `WasSuccessful` call. It continues to pay off in the future.